Added Commands for Climber (not currently finished, but skeleton is and most of inside)

DrSawickipedia opened the pull request.

DrSawickipedia committed:
Climber/Sonar Stuff

Created climber command, and methods for the climber. The sonar has been tested, and the scale factor found.

DrSawickipedia committed:
Comment Stuff

DrSawickipedia committed:
New Climber Stuff

DrSawickipedia committed:
Fixed Merge Errors

snikopoulos committed:
Merge with master after OI.java had a lot of while space changes

DrSawickipedia committed:
Created other commands for climber

DrSawickipedia committed:
Merge branch 'krs-climberStuff' of https://github.com/FRC4048/FRC2018_Java into krs-climberStuff

DrSawickipedia committed:
Added more to the commands, and changed from boolean flags to state machine (hopefully)

snikopoulos commented on src/org/usfirst/frc4048/commands/ClimbCommand.java:
```diff
@@ -0,0 +1,100 @@
+    	//SmartDashboard.putBoolean("Rung Detected:", Robot.climber.detectRung());
+    	
+    	if(Robot.climber.checkState(ClimbState.SetupDone) && Robot.climber.detectRung()) {
+    		Robot.arm.setExtensionToClimberPos();
```
What does setExtensionToClimberPos() do and who is writing the function? How does the command know when the "extension is at the climber position"?

snikopoulos commented on src/org/usfirst/frc4048/commands/ClimbCommand.java:
```diff
@@ -0,0 +1,100 @@
+    		Robot.arm.setExtensionToClimberPos();
+    			
+    	}else if(Robot.climber.checkState(ClimbState.SetupDone) && !Robot.climber.detectRung()) {
+    		DriverStation.reportError("Rung not detected!", true);
```
Perhaps the command should finish when this happens?

snikopoulos commented on src/org/usfirst/frc4048/subsystems/Arm.java:
```diff
@@ -281,6 +281,9 @@ public void setAutoExtension(boolean value)
     }
     
+    public void setExtensionToClimberPos() {
+    	//TODO Fill this in with actual code.
```
Have you talked with Matt about this function?

snikopoulos commented on src/org/usfirst/frc4048/commands/ClimbMakeReady.java:
```diff
@@ -0,0 +1,55 @@
+
+    // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=CONSTRUCTOR
+    public ClimbMakeReady() {
+    	if(Robot.climber.checkState(ClimbState.NothingDone)) {
```
I don't think you'll be able to have this check in the constructor. Or are you intending that the group command is created as part of a button press?

snikopoulos commented on src/org/usfirst/frc4048/subsystems/Climber.java:
```diff
@@ -38,7 +50,18 @@
+    private final double SCALE_FACTOR = 2.45; //Scale factor for the sonar (MB1240/20 = 2.45, MB1230 = 1.84(not verified by datasheet)) 
+   
+    //Constants for state of climber
+    private final int NOTHING_DONE = 0;
```
If you use the "climbState" member variable above, I don't think you'll need these int constants.

snikopoulos commented on src/org/usfirst/frc4048/subsystems/Climber.java:
```diff
@@ -58,7 +81,60 @@ public void periodic() {
+    
+    public boolean checkState(ClimbState climbState) {
+    	this.climbState = climbState;
+    	
```
This function can be changed to:
return this.climbState.equals(climbState)

No need for a case statement.

snikopoulos commented on src/org/usfirst/frc4048/subsystems/Climber.java:
```diff
@@ -58,7 +81,60 @@ public void periodic() {
+    public void changeState(ClimbState climbState) {
+    	this.climbState = climbState;
+    	
+    	switch(climbState) {
```
Same as above. No need to have a switch statement with all the cases. Storing the new climbState value is enough.

DrSawickipedia committed:
Got rid of unneeded constants, and fixed some other things

MatthewSpofford committed:
Re-added changes that were removed

DrSawickipedia committed:
Merge remote-tracking branch 'origin/master' into krs-climberStuff

Conflicts:
	src/org/usfirst/frc4048/OI.java
	src/org/usfirst/frc4048/Robot.java
	src/org/usfirst/frc4048/RobotMap.java
	src/org/usfirst/frc4048/subsystems/Arm.java

MatthewSpofford committed:
Added extension lowering for climbing

DrSawickipedia committed:
Merge remote-tracking branch 'origin/mes_tuesday_branch' into krs-climberStuff

Conflicts:
	src/org/usfirst/frc4048/commands/BlankCommand.java
	src/org/usfirst/frc4048/subsystems/Arm.java

DrSawickipedia committed:
Changed climber stuff into group commands only

There are two group commands for the climber. The first one (ClimbMakeReady), is what is used to move the arm into the right position, and then the driver would position the arm such that the rung limit switch is pressed, and the robot is able to climb. After, the operator would move the joystick to execute ClimbGroupCommand. The first time this is run, the rung limit switch must be pressed, or it will not do anything. If it is, the extension will go into the correct position, the climber pin will be pulled out, and the arm will move out of the way. Finally, after this, whenevr the joystick is moved, it will run ClimbCommand, which moves the winch normally.

DrSawickipedia committed:
Comment stuff

snikopoulos commented:
Closing without merging since we don't have a climber on the robot at the moment. Can bring back if we add a climber.
